File: backend/app/models/analysis.py
"""backend/app/models/analysis.py
Pydantic models for AI analysis data.
ML 출력과 프론트엔드가 공유하는 MVP Schema입니다.
"""
from typing import Literal
from pydantic import BaseModel


class Prediction(BaseModel):
    score: float
    level: Literal["LOW", "MEDIUM", "HIGH"]


class ModelInfo(BaseModel):
    name: str
    version: str


class GraphNode(BaseModel):
    id: str


class GraphEdge(BaseModel):
    source: str
    target: str


class Graph(BaseModel):
    nodes: list[GraphNode]
    edges: list[GraphEdge]


class RecommendedEdge(BaseModel):
    source: str
    target: str
    rank: int
    score_drop: float


class Validation(BaseModel):
    original_score: float
    recommended_removed_score: float
    random_removed_score: float


class AnalysisResult(BaseModel):
    """
    단일 분석 케이스의 전체 데이터.
    단일 온체인 조사 사건의 예측, 그래프, 추천 연결을 담습니다.
    """
    analysis_id: str
    dataset: str
    target_transaction: str
    prediction: Prediction
    model: ModelInfo
    graph: Graph
    recommended_edges: list[RecommendedEdge]
    validation: Validation
    status: str = "PENDING_REVIEW"


class CaseSummary(BaseModel):
    """
    Dashboard Case Card에 필요한 요약 정보.
    """
    analysis_id: str
    target_transaction: str
    prediction_score: float
    prediction_level: str
    model_name: str
    status: str

    model_config = {"protected_namespaces": ()}

File: frontend/src/types/analysis.ts
// frontend/src/types/analysis.ts
// ML 출력, 백엔드 API와 공유하는 MVP 분석 데이터 타입.

export interface Prediction {
  score: number;
  level: "LOW" | "MEDIUM" | "HIGH";
}

export interface ModelInfo {
  name: string;
  version: string;
}

export interface GraphNode {
  id: string;
}

export interface GraphEdge {
  source: string;
  target: string;
}

export interface Graph {
  nodes: GraphNode[];
  edges: GraphEdge[];
}

export interface RecommendedEdge {
  source: string;
  target: string;
  rank: number;
  score_drop: number;
}

export interface Validation {
  original_score: number;
  recommended_removed_score: number;
  random_removed_score: number;
}

export interface AnalysisResult {
  analysis_id: string;
  dataset: string;
  target_transaction: string;
  prediction: Prediction;
  model: ModelInfo;
  graph: Graph;
  recommended_edges: RecommendedEdge[];
  validation: Validation;
  status: string;
}

export interface CaseSummary {
  analysis_id: string;
  target_transaction: string;
  prediction_score: number;
  prediction_level: "LOW" | "MEDIUM" | "HIGH";
  model_name: string;
  status: string;
}
